### DISCUSSION TITLE
Shadowing oshi will result in the properties being renamed

### DISCUSSION THREAD

**Discussion by sorin-florea** [oshi/oshi#2633] [General]
Using [Gradle shadow](https://github.com/johnrengelman/shadow), relocating `oshi` to `com.relocated.oshi` will result in 

```String str = GlobalConfig.get("oshi.util.proc.path", "/proc");```

being changed to 

```String str = GlobalConfig.get("com.relocated.oshi.util.proc.path", "/proc");```

This is wrong since the properties will not be renamed. Is there any workaround for this issue?

**Comment by dbwiddis** [comment]
Hey @sorin-florea, thanks for raising this issue. I'm not sure I understand the full problem here.

OSHI's properties are named as strings in the code which are not tied to the project structure in any programmatic way.

It seems from your report that Gradle Shadow renames these strings based on assumptions about package structure.  While generally a reasonable default, it's clearly wrong in this case.  If there's a workaround, I would expect that Gradle Shadow would have some ability to disable or override this automatic assumption.

Open to any changes needed, but at first glance this does not seem to be OSHI's problem to solve.

**Reply by facewindu** [reply]
There is a way in the Gradle shadow plugin to exclude some files (like oshi `*.properties`) from being relocated.
And also a way to remap keys in property files, should they be looking like class names (that are relocated), via `PropertiesFileTransformer`.
All in all, there is a workaround to properly shadow oshi.
